### DISCUSSION TITLE
Toasts: SendOnNextPage()

### DISCUSSION THREAD

**Discussion by MGeurts** [tasteui/tasteui#394] [Q&A] [answered]
At the moment I'm still using a seperate package for toasts: [https://github.com/usernotnull/tall-toasts/](https://github.com/usernotnull/tall-toasts/) .

I want to switch to the TallStackUI toasts but I'm missing one feature: being able to **push a message on to the next page**.
This is very usefull when saving data in a form triggers a redirect to another (next) page.
In this case the TallStackUI toast appears only brief because it is interrupted by the subsequent redirect.

**Comment by devajmeireles** [comment, marked as answer]
Hey, @MGeurts !

Although many of our components work fine out of Livewire context, TallStackUI was built with Livewire applications in mind, so the way to interact with Toast and Dialog is through reactivity involving Livewire and AlpineJS. There is no native support for doing this, but perhaps you could try to achieve this result with the following (untested) logic:

1. Instead of dispatching a Toast, persist an array of data into the session when calling a public method
2. Redirects the user to another page, **that is a Livewire component**
3. Use the `mount` method of the Livewire component to check if the session exists and if yes, dispatch the Toast with the data.

If it works with Toast, will also work with Dialog.

**Comment by MGeurts** [comment]
I am using the toasts mostly inside the save methods of my Livewire components as in the example below.
The method saves the data, pushes the toast and then redirects back to the main pages (non livewire) .
Your suggestion would mean I have to use livewire components all over the application just to recieve events...

`
namespace App\Livewire\People\Edit;

use App\Livewire\Forms\People\PartnerForm;
use App\Models\Couple;
use App\Models\Person;
use Livewire\Component;

class Partner extends Component
{
    // -----------------------------------------------------------------------
    public $person;

    public $couple;

    public PartnerForm $partnerForm;

    // -----------------------------------------------------------------------
    public function mount()
    {
		...
    }

    public function savePartner()
    {
        if ($this->isDirty()) {
            $validated = $this->partnerForm->validate();

            $this->couple->update([
                'person1_id' => $this->person->id,
                'person2_id' => $validated['person2_id'],
                'date_start' => $validated['date_start'] ? $validated['date_start'] : null,
                'date_end' => $validated['date_end'] ? $validated['date_end'] : null,
                'is_married' => $validated['is_married'],
                'has_ended' => ($validated['date_end'] or $validated['has_ended']) ? true : false,
            ]);

            toast()->success(__('app.saved') . '.', __('app.saved'))->pushOnNextPage();
			
            $this->redirect('/people/' . $this->person->id);
        }
    }

    public function resetPartner()
    {
        $this->mount();
    }

    public function isDirty()
    {
        return ...
    }

    public function render()
    {
        ...

        return view('livewire.people.edit.partner')->with(compact('couple', 'persons'));
    }
}

`

**Reply by devajmeireles** [reply]
We have javascript API to display toast and dialog through Javascript, without relying on Livewire. That way, you can still display the toast as I mentioned. Simply send the array of the controller session to a Blade file and in the Blade file load the toast using the javascript API with the data from the array.
